runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/stableinput/BufferingDoFnRunner.java

```diff
@@ -125,8 +129,8 @@ public void processElement(WindowedValue<InputT> elem) {
       Instant outputTimestamp,
       TimeDomain timeDomain) {
     currentBufferingElementsHandler.buffer(
-        new BufferedElements.Timer(
-            timerId, timerFamilyId, window, timestamp, outputTimestamp, timeDomain));
+        new BufferedElements.Timer<>(
+            timerId, timerFamilyId, key, window, timestamp, outputTimestamp, timeDomain));
```

Comment:
On a closer look, I think adding to the Coder is the wrong way to preserve the key. DoFnOperator has a stateInternals() which should provide access to the current key, and that way we don't have to encode the key again.

Sorry for the bad suggestion.

Reply:
@reuvenlax  done